### PR TITLE
feat(multitenant): add certificate

### PR DESCRIPTION
**Pull request by apeabody**

### CODE REVIEW

File: 6-appsource/cymbal-bank/frontend/k8s/overlays/production/gateway.yaml [hunk 1]
```diff
@@ -31,6 +29,10 @@ spec:
     allowedRoutes:
       kinds:
       - kind: HTTPRoute
+    tls:
+      mode: Terminate
+      options:
+        networking.gke.io/pre-shared-certs: frontend-example-com
```
Comment (yliaog) [inline 6-appsource/cymbal-bank/frontend/k8s/overlays/production/gateway.yaml:35]: how is frontend.example.com converted to frontend-example-com ?

File: 2-multitenant/terraform.tfvars [hunk 2]
```diff
@@ -19,5 +19,8 @@ apps = {
     "ip_address_names" : [
       "frontend-ip",
     ]
+    "certificates" : {
+      "frontend" : ["frontend.example.com"]
```
Comment (yliaog) [inline 2-multitenant/terraform.tfvars:23]: should the key be "frontend-example-com" ?

File: 2-multitenant/modules/env_baseline/certs.tf [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    for key, value in var.apps : {
+      for k, v in value.certificates : "${key}-${k}" => v
+    }
+  ]...)
+
+  name    = each.key
+  project = data.google_project.eab_cluster_project.project_id
+
+  managed {
+    domains = each.value
```
Comment (yliaog) [inline 2-multitenant/modules/env_baseline/certs.tf:29]: how does Google managed cert verify the domain ? the DNS like https://cymbal-bank.endpoints.eab-gke-development-11d0.cloud.goog/  is verifiable by Google managed cert. Other DNS setup may require extra verification steps.

File: 6-appsource/cymbal-bank/frontend/k8s/overlays/production/gateway.yaml (hunk 1 above)
Comment (apeabody) [reply 6-appsource/cymbal-bank/frontend/k8s/overlays/production/gateway.yaml:35]: Yeah, it should be the `name` of the `google_compute_managed_ssl_certificate` which is `key` value from `certificates`.

File: 2-multitenant/modules/env_baseline/certs.tf (hunk 3 above)
Comment (apeabody) [reply 2-multitenant/modules/env_baseline/certs.tf:29]: Yeah, the current `frontend.example.com` value is basically a placeholder to demonstrate the plumbing, but we will eventually need a value which will work in the CI.

"This resource will "return" as soon as the certificate object is created, but post-creation the certificate object will go through a "provisioning" process. The provisioning process can complete only when the domain name for which the certificate is created points to a target pool which, itself, points at the certificate."

File: 2-multitenant/terraform.tfvars (hunk 2 above)
Comment (apeabody) [reply 2-multitenant/terraform.tfvars:23]: Yeah - or alternatively `networking.gke.io/pre-shared-certs` should just be `frontend`.  For now I'll just make them both `frontend-example-com` as eventually we'll need a valid value as this is a placeholder.

File: 2-multitenant/modules/env_baseline/certs.tf (hunk 3 above)
Comment (yliaog) [reply 2-multitenant/modules/env_baseline/certs.tf:29]: ok, sgtm

**Review by yliaog — APPROVED** [review]

**Review by amandakarina — APPROVED** [review]

**Review by gtsorbo — APPROVED** [review]

**Pull request merged**